File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys
input = sys.stdin.readline

n = int(input())
record = set()
answer = 0

for _ in range(n):
    string = input().strip()
    if string == "ENTER":
        record = set()
    else:
        if string not in record:
            answer+=1
        record.add(string)

print(answer)

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.cc
#include <bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n;
    cin >> n;
    
    set<string> record;
    int answer = 0;
    
    for (int i=0;i<n;i++){
        string s;
        cin >> s;
        if (s == "ENTER"){
            record.clear();
        }
        else{
            if (record.find(s) == record.end()){
                record.insert(s);
                answer++;
            }
        }
    }
    
    cout << answer;
    
    return 0;
}